cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsReporterSampler.java

```diff
@@ -101,36 +91,30 @@ public Samples getSamples(Cluster cluster,
           continue;
         }
         long recordTime = record.value().time();
-        if (recordTime + _acceptableMetricRecordProduceDelayMs < startTimeMs) {
+        if (recordTime + _acceptableMetricRecordProduceDelayMs < metricSamplerOptions.startTimeMs()) {
           LOG.debug("Discarding metric {} because its timestamp is more than {} ms earlier than the start time of sampling period {}.",
-                    record.value(), _acceptableMetricRecordProduceDelayMs, startTimeMs);
-        } else if (recordTime >= endTimeMs) {
+              record.value(), _acceptableMetricRecordProduceDelayMs, metricSamplerOptions.startTimeMs());
+        } else if (recordTime >= metricSamplerOptions.endTimeMs()) {
           TopicPartition tp = new TopicPartition(record.topic(), record.partition());
           LOG.debug("Saw metric {} whose timestamp is larger than the end time of sampling period {}. Pausing "
-                    + "partition {} at offset {}.", record.value(), endTimeMs, tp, record.offset());
+              + "partition {} at offset {}.", record.value(), metricSamplerOptions.endTimeMs(), tp, record.offset());
           partitionsToPause.add(tp);
         } else {
-          _metricsProcessor.addMetric(record.value());
+          addMetric(record.value());
           totalMetricsAdded++;
         }
       }
       if (!partitionsToPause.isEmpty()) {
         _metricConsumer.pause(partitionsToPause);
         partitionsToPause.clear();
       }
-    } while (!consumptionDone(_metricConsumer, endOffsets) && System.currentTimeMillis() < timeout);
+    } while (!consumptionDone(_metricConsumer, endOffsets) &&
+        System.currentTimeMillis() < metricSamplerOptions.timeoutMs());
     LOG.info("Finished sampling for topic partitions {} in time range [{},{}]. Collected {} metrics.",
-             _currentPartitionAssignment, startTimeMs, endTimeMs, totalMetricsAdded);
+        _currentPartitionAssignment, metricSamplerOptions.startTimeMs(),
+        metricSamplerOptions.endTimeMs(), totalMetricsAdded);
```

Comment:
Nit: Bad indentation -- the code style aligns when multiline.

Reply:
Fixed.